File: eon_timer/timers/second_timer.py
from typing import Final

from eon_timer import timers
from eon_timer.util.injector import component
from .calibrator import Calibrator


@component()
class SecondTimer:
    def __init__(self, calibrator: Calibrator) -> None:
        self.calibrator: Final = calibrator

    def create(self,
               target_second: int,
               calibration: float) -> list[float]:
        return [timers.to_minimum_length(target_second * 1000 + calibration + 200, self.calibrator.minimum_length)]

    def calibrate(self,
                  target_second: int,
                  second_hit: int) -> float:
        if second_hit < target_second:
            return ((target_second - second_hit) * 1000) - 500
        elif second_hit > target_second:
            return ((target_second - second_hit) * 1000) + 500
        else:
            return 0

File: src/timers/secondTimer.ts
import { toMinimumLength, MINIMUM_LENGTH } from '../utils/constants';

export function createSecondPhases(
  targetSecond: number,
  calibration: number,
  minimumLength: number = MINIMUM_LENGTH,
): number[] {
  return [toMinimumLength(targetSecond * 1000 + calibration + 200, minimumLength)];
}

export function calibrateSecond(targetSecond: number, secondHit: number): number {
  if (secondHit < targetSecond) {
    return (targetSecond - secondHit) * 1000 - 500;
  } else if (secondHit > targetSecond) {
    return (targetSecond - secondHit) * 1000 + 500;
  }
  return 0;
}
